konan_sdk/endpoints/base_endpoint.py

```diff
@@ -90,33 +109,10 @@ def post(self, request_object: ReqT) -> ResT:
         endpoint_request = self.prepare_request(request_object)
 
         logger.debug(f"Sending {self.name} request")
-        response = requests.post(
```

Comment:
This won't work I believe, need to call it like `self.endpoint_operation()(...` but how about we do it like this instead: define the request function to take in `method` like `def request(self, method: str, request_object: ReqT) -> ResT:` and in the body call it like `getattr(requests, method)(...)`, this way we could have an endpoint support both "get" / "post" / etc requests as long as the `prepare_request` and `process_response` handles it ( might have to add method param to them too ), what do you think of this?

Reply:
I had already tested that the original approach works before implementing it .. See the attached screenshot for an example.

![image](https://user-images.githubusercontent.com/91250709/149735328-0244fbd9-206e-43c8-b57e-6cc189374ed9.png)

About supporting multiple operations in the same class: I don't really like that idea because in some situations, the two (or more) operations will expect and return completely different objects with completely different "schemas". Therefore, just the two classes `ReqT` and `ResT` won't be sufficient. We will then have to have multiple `TypeVars`: `ReqT_GET`, `ResT_GET`, `ReqT_DELETE`, `ResT_DELETE`. etc .. 
Also, not all "child" `*Endpoint` classes will expose the same operations (`GET`, `POST`, `DELETE`, etc ...) and so for most of them, a lot of those extra, operation-scoped `TypeVars` will be useless.

Of course, we can completely do away with those `TypeVars` and rely on the `prepare_request` and `process_response` methods reading an additional `method` argument and dealing with the objects accordingly, but then we would lose the automatic type-hinting we get across different "child" `*Endpoint` classes.

Let me know if you find this justification sufficient or it needed additional modification.

Reply:
Ahh my bad, I just noticed its a property and not a method hence it will work 👍.  I agree, better to have automatic type hinting than a feature that might not be really necessary for grouping